example/gmt/sphere_preprocessing.c

```diff
@@ -0,0 +1,525 @@
+/*
+  This file is part of p4est.
+  p4est is a C library to manage a collection (a forest) of multiple
+  connected adaptive quadtrees or octrees in parallel.
+
+  Copyright (C) 2010 The University of Texas System
+  Additional copyright (C) 2011 individual authors
+  Written by Carsten Burstedde, Lucas C. Wilcox, and Tobin Isaac
+
+  p4est is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  p4est is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with p4est; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file sphere_preprocessing.c
```

Comment:
Thanks; nice description. If we were to include this program as one of our examples using the doxygen system, this can be done similarly like @SAutum has done it for `example/simple`.

To have the user run the program, would you like to include a small .csv input file that makes a friendly hello world type picture?

As for the main routine, would it make sense to optionally provide a second command line argument as name of the output file when it is desired to be written?

Reply:
>Thanks; nice description. If we were to include this program as one of our examples using the doxygen system, this can be done similarly like @SAutum has done it for example/simple.

Good idea. In my second PR there's more complete usage documentation that I think would belong in that location. I will include it when we get to that PR.

>To have the user run the program, would you like to include a small .csv input file that makes a friendly hello world type picture?

Can do!

>As for the main routine, would it make sense to optionally provide a second command line argument as name of the output file when it is desired to be written?

It complicates the options handling a bit, but I think there is probably a use case for it so I will do it.

Reply:
I hope this would be benign enough, when not given we do not write any output.

On a related note, would it make sense to have this example do nothing on any MPI rank > 0? In future incarnations of course, parallel operation will be implemented.

Reply:
> On a related note, would it make sense to have this example do nothing on any MPI rank > 0? In future incarnations of course, parallel operation will be implemented.

Sure.